services/bots/src/github-webhook/handlers/validate-cla.ts

```diff
@@ -0,0 +1,269 @@
+import { ConfigService } from '@nestjs/config';
+import { BaseWebhookHandler } from './base';
+
+import { DynamoDB } from 'aws-sdk';
+
+const ignoredAuthors: Set<string> = new Set([
+  // Ignore bot accounts that are not masked as bots
+  'travis@travis-ci.org',
+  'ImgBotHelp@gmail.com',
+  'support@lokalise.com',
+]);
+
+const ignoredRepositories: Set<string> = new Set([
+  // Ignore repositories that does not contain code
+  'home-assistant/home-assistant.io',
+  'home-assistant/assets',
+]);
+
+export class ValidateCla extends BaseWebhookHandler {
+  private ddbClient: DynamoDB;
+
+  constructor(configService: ConfigService) {
+    super(configService);
+    this.ddbClient = new DynamoDB({ region: 'us-west-2' });
+  }
+
+  async handle(eventType: string, eventData: Record<string, any>) {
+    if (!eventType.startsWith('pull_request')) {
+      return;
+    }
+
+    if (ignoredRepositories.has(eventData.repository.full_name)) {
+      return;
+    }
+
+    if (!['synchronize', 'opened', 'labeled'].includes(eventData.action)) {
+      return;
+    }
+    if (eventData.action === 'labeled') {
+      if (eventData.label.name !== 'cla-recheck') {
+        return;
+      } else if (eventData.label.name === 'cla-recheck') {
+        await this.githubApiClient.issues.removeLabel({
+          owner: eventData.repository.owner.login,
+          repo: eventData.repository.name,
+          issue_number: eventData.number,
+          name: 'cla-recheck',
+        });
+      }
+    }
+
+    const commits = await this.getCommits(eventData);
+    await this.checkAuthors(eventData, commits.data);
+  }
+
+  getCommits(eventData: Record<string, any>) {
+    return this.githubApiClient.pulls.listCommits({
+      owner: eventData.repository.owner.login,
+      repo: eventData.repository.name,
+      pull_number: eventData.number,
+      per_page: 100,
+    });
+  }
+
+  async checkAuthors(eventData: Record<string, any>, commits: any[]) {
+    const authorsWithSignedCLA: Set<string> = new Set();
+    const authorsNeedingCLA: { sha: string; login: string }[] = [];
+    const commitsWithoutLogins: { sha: string; maybeText: string; maybeEmailAddress: boolean }[] =
+      [];
+
+    for await (const commit of commits) {
+      if (commit?.author?.type === 'Bot' || ignoredAuthors.has(commit?.commit?.author?.email)) {
+        return;
+      }
+
+      if (!commit?.author && commit?.commit?.author?.email) {
+        const maybeCheck = commit?.commit?.author?.email?.indexOf('@') > -1;
```

Comment:
You don't need the `?` as you already checked those in the if-statment. Also you can use `includes`
```suggestion
        const maybeCheck = commit.commit.author.email.includes('@');
```